packages/replay-next/src/hooks/useDebounce.ts

```diff
@@ -13,5 +13,10 @@ export function useDebounce<Value>(value: Value, delay: number = 100): Value {
     };
   }, [value, delay]);
 
+  return [debouncedValue, setDebouncedValue] as const;
```

Comment:
I don't think we should make this change. The purpose of this hook is to debounce a value automatically. Calling the state updater function externally– it's no longer debounced and potentially introduces a race case:

1. New value A passed as param to `useDebouncedState` which starts a debounce timer
2. `setDebouncedValue` state updater is called from an external function with value B
3. Timer fires and overrides with "debounced" value A

I think maybe you want a combination of debounce+transition here. (I don't think we have a shared hook for this, but maybe we don't need it right away?) Something like this (I didn't test this)...

```js
const [valueForDisplay, setValueForDisplay] = useState("");
const [valueForSuspense, setValueForSuspense] = useState("");
const [isPending, startTransition] = useTransition();

useEffect(() => {
  const handler = setTimeout(() => {
    startTransition(() => {
      setDebouncedValue(value);
    });
  }, delay);

  return () => {
    clearTimeout(handler);
  };
}, [valueForSuspense]);
```

Then you'd pass `valueForSuspense` to your Suspense cache and maybe dim the UI or something else when `isPending` is true (to show you're updating results)

Reply:
I see, hmm but maybe we don't need to keep "search on enter" functionality either(?) in which case we can just keep the automatically debounced value, without having any need to update this value without skipping the timeout

Reply:
> maybe we don't need to keep "search on enter" functionality either(?)

Totally agreed. We can drop that.

The addition of `useTransition` would be nice (but you can approximate that by e.g. `const isPending = queryForDisplay !== queryForSearch`)